File: ios/SystemInfo.m
//
//  SystemInfo.m
//  DietDoctor
//
//  Created by Kassem Itani on 10/26/20.
//
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

// The class that contain your methods
@interface RCT_EXTERN_MODULE(SystemInfo, NSObject)

// The methods you are exposing to Javascript
RCT_EXTERN_METHOD(getLanguage: (RCTResponseSenderBlock)callback);
RCT_EXTERN_METHOD(getBundleIdentifier: (RCTResponseSenderBlock)callback);

@end

File: ios/SystemInfo.swift
//
//  SystemInfo.swift
//  DietDoctor
//
//  Created by Kassem Itani on 10/26/20.
//

import Foundation
import React

// add @obj modifier here
@objc(SystemInfo)
class SystemInfo: NSObject {
  
  @objc func getLanguage( _ callback: @escaping RCTResponseSenderBlock){
    let langStr = Locale.current.languageCode
    callback([langStr ?? ""])
  }
  
  @objc func getBundleIdentifier( _ callback: @escaping RCTResponseSenderBlock) {
    let bundleID = Bundle.main.bundleIdentifier
    callback([bundleID ?? ""])
  }
}
